dhctl/pkg/kubernetes/actions/deckhouse/delete.go

```diff
@@ -344,10 +355,109 @@ func DeleteMachinesIfResourcesExist(kubeCl *client.KubernetesClient) error {
 		return fmt.Errorf("Machines deletion aborted.\n")
 	}
 
-	err = DeleteMachineDeployments(kubeCl)
+	err = DeleteMCMMachineDeployments(kubeCl)
 	if err != nil {
 		return err
 	}
 
-	return WaitForMachinesDeletion(kubeCl)
+	if err := WaitForMCMMachinesDeletion(kubeCl); err != nil {
+		return err
+	}
+
+	// try to remove CAPI machines it needs for static clusters and cluster with cluster api support
+	err = retry.NewLoop("Get Kubernetes cluster resources for CAPI group/version", 5, 5*time.Second).WithShowError(false).
+		Run(func() error {
+			return checkCAPIMachinesAPI(kubeCl)
+		})
+	if err != nil {
+		log.WarnF("Can't get resources in group=cluster.x-k8s.io, version=v1beta1: %v\n", err)
+		if input.NewConfirmation().
+			WithMessage("Machines weren't deleted from the cluster. Do you want to continue?").
+			WithYesByDefault().
+			Ask() {
+			return nil
+		}
+		return fmt.Errorf("Machines deletion aborted.\n")
+	}
+
+	err = DeleteCAPIMachineDeployments(kubeCl)
+	if err != nil {
+		return err
+	}
+
+	return WaitForCAPIMachinesDeletion(kubeCl)
+}
+
+// CAPI
+const (
+	CAPIGroup        = "cluster.x-k8s.io"
+	CAPIGroupVersion = "v1beta1"
+)
+
+var capiMachinesSchema = schema.GroupVersionResource{Group: CAPIGroup, Version: CAPIGroupVersion, Resource: "machines"}
+
+func checkCAPIMachinesAPI(kubeCl *client.KubernetesClient) error {
+	gv := schema.GroupVersion{
+		Group:   CAPIGroup,
+		Version: CAPIGroupVersion,
+	}
+
+	return checkMachinesAPI(kubeCl, gv)
+}
+
+func DeleteCAPIMachineDeployments(kubeCl *client.KubernetesClient) error {
+	machineDeploymentsSchema := schema.GroupVersionResource{Group: CAPIGroup, Version: CAPIGroupVersion, Resource: "machinedeployments"}
+
+	return retry.NewLoop("Delete CAPI MachineDeployments", 45, 5*time.Second).Run(func() error {
+		allMachines, err := kubeCl.Dynamic().Resource(capiMachinesSchema).Namespace(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("get machines: %v", err)
+		}
+
+		for _, machine := range allMachines.Items {
+			m := machine
+			unstructured.SetNestedField(m.Object, "10s", "spec", "nodeDrainTimeout")
```

Comment:
Are you sure such a small timeout is enough?

Reply:
needs comment

Reply:
done